builtin/mainmenu/settings/dlg_file_browser.lua

```diff
@@ -0,0 +1,160 @@
+-- Minetest
+-- Copyright (C) 2020 Jean-Patrick Guerrero <jeanpatrick.guerrero@gmail.com>
+-- Copyright (C) 2020 EvidenceBKidscode
+-- Copyright (C) 2023 ROllerozxa
+--
+-- This program is free software; you can redistribute it and/or modify
+-- it under the terms of the GNU Lesser General Public License as published by
+-- the Free Software Foundation; either version 2.1 of the License, or
+-- (at your option) any later version.
+--
+-- This program is distributed in the hope that it will be useful,
+-- but WITHOUT ANY WARRANTY; without even the implied warranty of
+-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+-- GNU Lesser General Public License for more details.
+--
+-- You should have received a copy of the GNU Lesser General Public License along
+-- with this program; if not, write to the Free Software Foundation, Inc.,
+-- 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+local PATH = os.getenv("HOME") -- Linux & similar
+		or os.getenv("HOMEDRIVE") .. os.getenv("HOMEPATH") -- Windows
+		or core.get_user_path() -- If nothing else works
```

Comment:
shouldn't this be in dialogdata? why is it global?

Reply:
Also, falling back to `core.get_user_path()` doesn't work.

```
2023-06-06 19:26:46: ERROR[Main]: GUIEngine: execution of menu script failed: Failed to load and run script from /minetest/bin/../builtin/init.lua:
2023-06-06 19:26:46: ERROR[Main]: ...st/bin/../builtin/mainmenu/settings/dlg_file_browser.lua:21: attempt to concatenate a nil value
2023-06-06 19:26:46: ERROR[Main]: stack traceback:
2023-06-06 19:26:46: ERROR[Main]: 	...st/bin/../builtin/mainmenu/settings/dlg_file_browser.lua:21: in main chunk
2023-06-06 19:26:46: ERROR[Main]: 	[C]: in function 'dofile'
2023-06-06 19:26:46: ERROR[Main]: 	.../minetest/bin/../builtin/mainmenu/settings/init.lua:23: in main chunk
2023-06-06 19:26:46: ERROR[Main]: 	[C]: in function 'dofile'
2023-06-06 19:26:46: ERROR[Main]: 	.../minetest/bin/../builtin/mainmenu/init.lua:43: in main chunk
2023-06-06 19:26:46: ERROR[Main]: 	[C]: in function 'dofile'
2023-06-06 19:26:46: ERROR[Main]: 	/minetest/bin/../builtin/init.lua:63: in main chunk
2023-06-06 19:26:46: ERROR[Main]: No future without main menu!
```

Reply:
I presume it is in order to remember the last path you had navigated to, for the duration of the session? Though it might make more sense to just start at the home directory always.